executor/grant_test.go

```diff
@@ -189,7 +189,7 @@ func (s *testSuite3) TestNoAutoCreateUser(c *C) {
 	tk.MustExec(`SET sql_mode='NO_AUTO_CREATE_USER'`)
 	_, err := tk.Exec(`GRANT ALL PRIVILEGES ON *.* to 'test'@'%' IDENTIFIED BY 'xxx'`)
```

Comment:
em... I execute this success in mysql 5.7. 
In mysql doc: https://dev.mysql.com/doc/refman/5.7/en/sql-mode.html#sqlmode_no_auto_create_user,
> Prevent the GRANT statement from automatically creating new user accounts if it would otherwise do so, **unless authentication information is specified**.

Reply:
SET sql_mode='NO_AUTO_CREATE_USER'
SET sql_mode=''

The behavior is different, have a try! @crazycs520